crates/hdk/src/hash.rs

```diff
@@ -1,3 +1,109 @@
+//! Functions to generate standardized hashes of Holochain elements and arbitrary bytes.
+//!
+//! Holochain makes extensive use of hashes to address any content. It utilizes [Blake2b](https://www.blake2.net/) as hashing algorithm.
+//! Hashes in Holochain have a length of 39 bytes, made up of 3 bytes for identifying the hash and 36 bytes of digest.
+//!
+//! ## Example
+//! This is an example of a public agent key hash, displayed as a byte array in decimal notation:
+//!
+//! ```
+//! 132  32  36  39 218 126  34  87
+//! 204 165 227 255  29 236 160  66
+//! 221 163 168 112 215 187 143 152
+//!  68   4  30 206 173 203 210 111
+//! 103 207 124   2 107  67  33
+//! ```
+//!
+//! ### Base64 encoding
+//!
+//! Since hashes have to be exchanged over the network, their bytes are encoded before sending and decoded after reception,
+//! to avoid data corruption during transport. To convert the hashes from a byte format to a transferrable string, Holochain encodes them with the
+//! [Base64 scheme](https://developer.mozilla.org/en-US/docs/Glossary/Base64). Encoding the example public agent key in Base64 results in:
+//!
+//! ```
+//! hCAkJ9p+IlfMpeP/HeygQt2jqHDXu4+YRAQezq3L0m9nz3wCa0Mh
+//! ```
+//!
+//! ### Self-identifying hash encoding
+//!
+//! Following the [Multibase protocol](https://github.com/multiformats/multibase), hashes in Holochain self-identify its encoding in Base64.
+//! All hashes in text format are prefixed with a `u`, to identify them as [`base64url`](https://github.com/multiformats/multibase/blob/master/multibase.csv#L23).
```

Comment:
this line is 162 characters long... our rustfmt currently uses the default max_width of 100 characters (but alas it doesn't reformat comments), I personally use an editor with a width of 80 characters.

I vote for keeping comments generally under at least 100 characters with occasional exceptions for urls, etc. And I personally try to keep them under 80.

Reply:
Sure, I hard wrapped the doc comments at 80 chars.